evm/vm/base.py

```diff
@@ -300,6 +321,13 @@ def generate_block_from_parent_header_and_coinbase(cls, parent_header, coinbase)
     # Validate
     #
     def validate_block(self, block):
+        if not isinstance(block, self.get_block_class()):
+            raise ValidationError(
+                "This vm ({0!r}) is not equipped to validate a block of type {1!r}".format(
+                    self,
+                    block,
+                )
+            )
```

Comment:
I guess this check will be moot, when this method moves up to Chain. But we are a couple days from moving this one, I think.